sfge/src/main/java/com/salesforce/rules/unusedmethod/ConstructorMethodCallValidator.java

```diff
@@ -0,0 +1,83 @@
+package com.salesforce.rules.unusedmethod;
+
+import com.salesforce.graph.vertex.*;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Helper class for {@link com.salesforce.rules.UnusedMethodRule}. Used for determining whether a
+ * constructor is ever invoked, e.g., {@code this()}, {@code super()}, or {@code new Whatever()}.
+ */
+public class ConstructorMethodCallValidator extends BaseMethodCallValidator {
+
+    public ConstructorMethodCallValidator(
+            MethodVertex targetMethod, RuleStateTracker ruleStateTracker) {
+        super(targetMethod, ruleStateTracker);
+    }
+
+    /**
+     * For constructors, an "internal" invocation occurs when another constructor delegates to this
+     * one.
+     *
+     * <ol>
+     *   <li>Another constructor in the same class calling {@code this()}.
+     *   <li>A constructor in a DIRECT subclass calling {@code super()}. Constructors only inherit
+     *       one level.
+     * </ol>
+     *
+     * @return - True if constructor is invoked internally.
+     */
+    @Override
+    protected boolean internalUsageDetected() {
+        // First, check for usage in the class where the target method is defined.
```

Comment:
This code is a consolidation of code from `InternalCallValidator` and `SubclassCallValidator`.